### PR TITLE
chore(concept): #521 revoked notification handling

### PR DESCRIPTION
**Pull request by ds-crehm**
##Description
Add concept for https://github.com/eclipse-tractusx/traceability-foss/issues/521

resolves eclipse-tractusx/traceability-foss#521

### CODE REVIEW

File: docs/concept/#521-revoked-notification-handling/#521-revoked-notification-handling.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+# Overview
+
+After a notification is created and approved, relevant policies must be verified before it can be sent.
+There are three possibilities:
+1. The policy is valid (validUntil >= current DateTime) and the notification is permitted to be sent.
+2. The policy is valid (validUntil >= current DateTime) but the notification is **not** permitted to be sent.
+3. The policy is **not** valid (validUntil < current DateTime) and the notification is **not** permitted to be sent (policy details are not relevant in this case).
+
+Policies are stored in the IRS. The IRS provides an API to fetch and verify policies.
```
Comment (mkanal) [inline docs/concept/#521-revoked-notification-handling/#521-revoked-notification-handling.md:31]:
```suggestion
Policies are stored in the IRS's PolicyStore. The IRS provides an PolicyStore API to create, fetch and verify policies.
```

File: docs/concept/#521-revoked-notification-handling/#521-revoked-notification-handling.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+9. [Glossary](#glossary)
+10. [References](#references)
+11. [Additional Details](#additional-details)
+
+# Overview
+
+After a notification is created and approved, relevant policies must be verified before it can be sent.
+There are three possibilities:
+1. The policy is valid (validUntil >= current DateTime) and the notification is permitted to be sent.
+2. The policy is valid (validUntil >= current DateTime) but the notification is **not** permitted to be sent.
```
Comment (mkanal) [inline docs/concept/#521-revoked-notification-handling/#521-revoked-notification-handling.md:28]: Case 2 is not clear to me. There are two validity constraints, time based (policy expiration) and permission based (policy definition matched the policy of catalog offer)

File: docs/concept/#521-revoked-notification-handling/#521-revoked-notification-handling.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+3. The policy is **not** valid (validUntil < current DateTime) and the notification is **not** permitted to be sent (policy details are not relevant in this case).
+
+Policies are stored in the IRS. The IRS provides an API to fetch and verify policies.
+
+# Summary
+
+When the notification is not sent because of a permission issue or an invalid policy, the user must be notified properly.
```
Comment (mkanal) [inline docs/concept/#521-revoked-notification-handling/#521-revoked-notification-handling.md:35]: Permission issue in one constraint of an invalid policy 
invalid policy 
 - time based 
 - permission or constraint based

File: docs/concept/#521-revoked-notification-handling/#521-revoked-notification-handling.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+- During policy check, throw separate Exceptions based on the type of failure.
+  - If policy is not valid -> UsagePolicyExpiredException
+  - If policy is valid but notification not permitted -> UsagePolicyPermissionException
+- New quality investigation & alert status: "Failed"
+  - Notification set to this status, when it could not be sent due to the policy exceptions
+- Pop-up informing the user of the exception
+- Detailed status information stored in the message history of the notification
+- User must be able to resend the notification
+
+# NFR
```
Comment (mkanal) [inline docs/concept/#521-revoked-notification-handling/#521-revoked-notification-handling.md:51]:
```suggestion

```

File: docs/concept/#521-revoked-notification-handling/#521-revoked-notification-handling.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+# Out of scope
```
Comment (mkanal) [inline docs/concept/#521-revoked-notification-handling/#521-revoked-notification-handling.md:53]:
```suggestion

```

File: docs/concept/#521-revoked-notification-handling/#521-revoked-notification-handling.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        deactivate IRSLib
+        TraceX->>TraceX: Notification status = Failed
+        TraceX->>TraceX: Create error message
+    end
+```
+
+### Frontend
+
+After creating and approving the notification and one of the exceptions is thrown:
+1. A pop-up must be shown to the user
```
Comment (mkanal) [inline docs/concept/#521-revoked-notification-handling/#521-revoked-notification-handling.md:106]:
```suggestion
1. An error toast message  must be shown to the user
```

File: docs/concept/#521-revoked-notification-handling/#521-revoked-notification-handling.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+2. The notification status must be changed to "Failed"
```
Comment (mkanal) [inline docs/concept/#521-revoked-notification-handling/#521-revoked-notification-handling.md:107]: We have to think about a transient state model which extends the standardized notification model with additional states -. 

See https://eclipse-tractusx.github.io/traceability-foss/docs/user/user-manual.html Quality alert status this is standardized and could not be changed. There is no ERROR or FAILED state. We have to extend this official notification model with transient states which are only handled within the Trace-X application. Please provide a sample where FAILED will be inside the state diagram. Thanks

**Review by mkanal — CHANGES_REQUESTED** [review]
Hi Christian, 
great work. Please see my remarks inside this PR.

File: docs/concept/#521-revoked-notification-handling/#521-revoked-notification-handling.md [hunk 8]
```diff
@@ -5,53 +5,42 @@
...
-
 # Requirements
 
 - During policy check, throw separate Exceptions based on the type of failure.
   - If policy is not valid -> UsagePolicyExpiredException
   - If policy is valid but notification not permitted -> UsagePolicyPermissionException
 - New quality investigation & alert status: "Failed"
   - Notification set to this status, when it could not be sent due to the policy exceptions
-- Pop-up informing the user of the exception
+- Error text message informing the user of the exception
```
Comment (mkanal) [inline docs/concept/#521-revoked-notification-handling/#521-revoked-notification-handling.md:40]: Toast messages

File: docs/concept/#521-revoked-notification-handling/#521-revoked-notification-handling.md [hunk 9]
```diff
@@ -103,7 +95,7 @@ sequenceDiagram
 ### Frontend
 
 After creating and approving the notification and one of the exceptions is thrown:
-1. A pop-up must be shown to the user
+1. An error text message must be shown to the user
```
Comment (mkanal) [inline docs/concept/#521-revoked-notification-handling/#521-revoked-notification-handling.md:98]: Toast message

**Review by mkanal — CHANGES_REQUESTED** [review]

**Review by mkanal — APPROVED** [review]
LGFM

**Review by ds-mwesener — APPROVED** [review]

**Pull request merged**